### ISSUE TITLE
postgresql modules: allow users to pass inputs containting dots where possible 

### ISSUE THREAD

**Issue by Andersson007** [ansible-collections/community.general#210]
##### SUMMARY
postgresql modules:
1. allow users to pass inputs containting dots where possible
2. add trust_input parameter

Related to https://github.com/ansible-collections/community.general/issues/106

Already done for:
postgresql_user
postgresql_privs
postgresql_owner
postgresql_membership
postgresql_db
postgresql_tablespace
postgresql_schema

Can be implemented similar to
https://github.com/ansible-collections/community.general/pull/198

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
postgresql modules

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/database/postgresql`](https://github.com/ansible-collections/community.general/blob/master/plugins/modules/database/postgresql)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @Dorn- @amenonsen @andytom @jbscalia @kostiantyn-nemchenko @matburt @nerzhul @sebasmannem @tcraxs
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Andersson007** [comment]
if anybody has nothing to do and wants to help, there are several postges modules which need the same.
The list of already fixed modules is in the description, so there are 10+ left at the moment.
If there are volunteers, text me before starting please not to do the same stuff simultaneously.
If nobody wants/can make time, no problem:) it'll take a bit longer.

**Comment by andytom** [comment]
> if anybody has nothing to do and wants to help, there are several postges modules which need the same.
> The list of already fixed modules is in the description, so there are 10+ left at the moment.
> If there are volunteers, text me before starting please not to do the same stuff simultaneously.
> If nobody wants/can make time, no problem:) it'll take a bit longer.

I might have some free time next week so might be able to pick up a few module, is there a list of all modules still to be updated?

**Comment by Andersson007** [comment]
@andytom , thanks!

postgresql_ext.py
postgresql_idx.py
postgresql_lang.py
postgresql_publication.py
postgresql_query.py
postgresql_sequence.py
postgresql_set.py
postgresql_slot.py
postgresql_subscription.py
postgresql_table.py

I think only `session_role` for the following (but not 100% sure):
postgresql_user_obj_stat_info.py
postgresql_info.py
postgresql_ping.py
postgresql_copy.py

postgresql_pg_hba.py doesn't connect to db

**Comment by andytom** [comment]
I'll take a look at `postgresql_ext` next week and see how I get on.

**Comment by Andersson007** [comment]
nice, i'll take `postgresql_idx` then

**Comment by Andersson007** [comment]
Yesterday i finished `postgresql_lang.py`, i'll begin `postgresql_publication` now

**Comment by Andersson007** [comment]
finished `postgresql_publication` and `postgresql_subscription`

**Comment by Andersson007** [comment]
I'll start `postgresql_query`

**Comment by andytom** [comment]
I'll try to take a look at `postgresql_sequence` next.

**Comment by Andersson007** [comment]
i'll take a look at `postgresql_set`

**Comment by andytom** [comment]
I'll do `postgresql_slot` next.

**Comment by Andersson007** [comment]
i'll do postgresql_table.py now

**Comment by Andersson007** [comment]
@andytom i'll left `postgresql_user_obj_stat_info` for you, can you make time for that?

**Comment by Andersson007** [comment]
i'll do `postgresql_info` next

**Comment by Andersson007** [comment]
done

**Comment by Andersson007** [comment]
@andytom , so, would you like to do `postgresql_user_obj_stat_info `, `postgresql_ping`, `postgresql_copy` ? (no rush).
And that's it with this, i think.

**Comment by andytom** [comment]
Yep, I'll try to do them this afternoon.

**Comment by andytom** [comment]
* `postgresql_user_obj_stat_info` - #310 
* `postgresql_ping` - Only executes a static query (`SELECT version()`)

I'm probably not going to have time to look at `postgresql_copy`.

**Comment by Andersson007** [comment]
must be fixed now
@andytom thanks for taking part in this!
